### DISCUSSION TITLE
Missing push options --all --mirror

### DISCUSSION THREAD

**Discussion by janderssonse** [go-git/go-git#1054] [General]
A few push-options missing from go-git https://github.com/go-git/go-git/blob/5d08d3bd94c65a3b6c25c6fba6907d12b0dac4ca/options.go#L223 that could improve go-gitlang is added support for:

--all
--mirror

**Comment by pjbgf** [comment]
I am under the impression this would be achievable currently by setting the refspecs. Something like:

```go
    pushOpts := &git.PushOptions{
        RefSpecs: []config.RefSpec{
            config.RefSpec("+refs/*:refs/*"),
        },
        Force: true,
    }
    
    err = remote.PushContext(context.Background(), pushOpts)
    if err != nil {
        return err 
    }
```

But you may need to do an additional push to delete the remote refs that don't exist locally. The refspec for that may look like [this](https://github.com/go-git/go-git/blob/5d08d3bd94c65a3b6c25c6fba6907d12b0dac4ca/config/refspec_test.go#L57-L58).

**Comment by janderssonse** [comment]
Thanks, for the helpful workarounds, but still also hope to see  the options available in easy to use flags just like Force for example.

**Comment by pjbgf** [comment]
As a library, `go-git` aims to provide the same functionality as the upstream `git` CLI, which it does in this case, although with a different UX which requires users to understand what a specific `git` CLI does behind the scenes.

Given that the behaviour is already supported, I am moving this as a discussion so that we can gauge other opinions. Overall this could be tackle on several manners:

1. Try to align 1:1 upstream options with upstream.
2. Improve documentation to help users navigate `git` -> `go-git`.
3. Expand _examples so that users can see working example of `git` in `go-git`.

My preference would probably be the last one as introducing all flags from `git` to a library may lead to an awkward UX.

**Comment by wingsofovnia** [comment]
While I don't think aligning git & go-git API is really necessary as long as same things can be achieved, in my opinion, exposing handful shortcuts / "syntax sugar" via options is valuable and can't hurt. 

Moreover, in this particular example, there is already `CloneOptions#Mirror` option:
https://github.com/go-git/go-git/blob/d1843220b6ab920fd3f5509ee532799a6fe13f74/options.go#L49-L56

So adding `PushOptions#Mirror` will complement go-git existing API.
